platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +251,242 @@ def get_sfp(self, index):
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_dpu_thermal':
+                (ModuleBase.REBOOT_CAUSE_SW_THERMAL, ''),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_aux_pwr_or_reload':
+                (ModuleBase.REBOOT_CAUSE_POWER_LOSS, ''),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/reset_pwr_off':
+                (ModuleBase.REBOOT_CAUSE_DPU_SELF_REBOOT, ''),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/tpm_rst':
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'Reset by the TPM module'),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/perst_rst':
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'PERST# signal to ASIC'),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/phy_rst':
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'Phy reset'),
+            f'/run/hw-management/system/dpu{self._name.lower()}/system/usbphy_rst':
+                (ModuleBase.REBOOT_CAUSE_HARDWARE_OTHER, 'USB Phy reset'),
+        }
+        for f, rd in reboot_cause_map.items():
+            if utils.read_int_from_file(f) == 1:
+                return rd
+        return ModuleBase.REBOOT_CAUSE_NON_HARDWARE, ''
+
+    def get_midplane_ip(self):
+        """
+        Retrieves the midplane IP-address of the module in a modular chassis
+        When called from the Supervisor, the module could represent the
+        line-card and return the midplane IP-address of the line-card.
+        When called from the line-card, the module will represent the
+        Supervisor and return its midplane IP-address.
+        When called from the DPU, returns the midplane IP-address of the dpu-card.
+        When called from the Switch returns the midplane IP-address of Switch.
+        Returns:
+            A string, the IP-address of the module reachable over the midplane
+        """
+        midplane_data = DeviceDataManager.get_platform_midplane_network()
+        network_cidr = midplane_data['bridge_address']
+        ip_network_cidr = ip_network(network_cidr, strict=False)
+        return str(ip_network_cidr[self.dpu_id])
+
+    def is_midplane_reachable(self):
+        """
+        Retrieves the reachability status of the module from the Supervisor or
+        of the Supervisor from the module via the midplane of the modular chassis
+        Returns:
+            A bool value, should return True if module is reachable via midplane
+        """
+        command = ['ping', '-c', '1', '-W', '1', self.get_midplane_ip()]
+        try:
+            return subprocess.call(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL) == 0
+        except Exception:
+            return False
+
+    def _is_dataplane_int_online(self):
+        platform_dpus_data = DeviceDataManager.get_platform_dpus_data()
+        module_name = self.get_name()
+        for dictionary in platform_dpus_data:
```

Comment:
Why do you store the DPU data as a list? This should be a dictionary. Please change in all places

Reply:
Done